runzi/automation/scaling/toshi_api/aggregate_inversion_solution.py

```diff
@@ -0,0 +1,120 @@
+
+from datetime import datetime as dt
+from hashlib import md5
+from pathlib import PurePath
+
+import base64
+import json
+import requests
+
+from nshm_toshi_client.toshi_client_base import ToshiClientBase, kvl_to_graphql
+
+import logging
+log = logging.getLogger(__name__)
+#logging.basicConfig(level=logging.DEBUG)
+
+class AggregateInversionSolution(object):
+
+    def __init__(self, api):
+        self.api = api
+        assert isinstance(api, ToshiClientBase)
+
+    def upload_inversion_solution(self, task_id, filepath, source_solution_ids, aggregation_fn, mfd_table=None, meta=None, predecessors=None, metrics=None):
+        filepath = PurePath(filepath)
+        file_id, post_url = self._create_inversion_solution(filepath, task_id, source_solution_ids, aggregation_fn, mfd_table, meta, predecessors, metrics)
+        self.upload_content(post_url, filepath)
+
+        #link file to task in role
+        self.api.task_file.create_task_file(task_id, file_id, 'WRITE')
+        return file_id
+
+    def upload_content(self, post_url, filepath):
+        log.debug(f'upload_content() POST URL: {post_url}; PATH: {filepath}')
+        filedata = open(filepath, 'rb')
+        files = {'file': filedata}
+        log.debug(f'upload_content() _s3_url: {self.api._s3_url}')
+
+        response = requests.post(
+            url=self.api._s3_url,
+            data=post_url,
+            files=files)
+        log.debug(f'response {response}')
+        response.raise_for_status()
+
+    def _create_inversion_solution(self, filepath, produced_by, source_solution_ids, aggregation_fn, mfd_table=None, meta=None, predecessors=None, metrics=None):
+        qry = '''
```

Comment:
pls add this field (it will become mandatory soon) ... `common_rupture_set` which is the ID! of same.